tests/DCPS/Compiler/optional/main.cpp

```diff
@@ -6,11 +7,51 @@
 
 #include <ace/ACE.h>
 #include <ace/Log_Msg.h>
+#include <gtest/gtest.h>
 
-#include <cstdlib>
+TEST(OptionalTests, Empty)
+{
+  optional::OptionalMembers empty;
+  EXPECT_FALSE(empty.opt().has_value());
+  empty.opt(12);
+  EXPECT_TRUE(empty.opt().has_value());
+}
+
+TEST(OptionalTests, SerializationSize)
+{
+  optional::OptionalMembers empty;
+  OpenDDS::DCPS::Encoding encoding;
+  encoding.kind(OpenDDS::DCPS::Encoding::KIND_XCDR1);
+
+  EXPECT_EQ(3, OpenDDS::DCPS::serialized_size(encoding, empty));
```

Comment:
This doesn't seem right for XCDR1. I think each unset optional member in this struct occupies a 4-byte header with the value of the size in the header is 0. The total size of 3, however, looks right for XCDR2. So maybe there's some mixing between them here.

Reply:
I need to debug it further, but I added the XCDR2 test and the serialized_size is now returning 7 and not 3

This is the generated code for it, right now. I did somewhat overlook the difference between XCDR1&2. But presumably this should work for XCDR2?
```c++
void serialized_size(const Encoding& encoding, size_t& size, const ::optional::OptionalMembers& stru)
{
  ACE_UNUSED_ARG(encoding);
  ACE_UNUSED_ARG(size);
  ACE_UNUSED_ARG(stru);
  if (encoding.xcdr_version() == Encoding::XCDR_VERSION_2) {
    serialized_size_delimiter(encoding, size);
  }
  primitive_serialized_size(encoding, size, ACE_OutputCDR::from_boolean(stru.opt().has_value()));
  if (stru.opt().has_value()) {
  primitive_serialized_size(encoding, size, stru.opt().value());
  }
  primitive_serialized_size(encoding, size, ACE_OutputCDR::from_boolean(stru.strOpt().has_value()));
  if (stru.strOpt().has_value()) {
  primitive_serialized_size_ulong(encoding, size);
  size += stru.strOpt().value().size() + 1;
  }
  primitive_serialized_size(encoding, size, ACE_OutputCDR::from_boolean(stru.seqOpt().has_value()));
  if (stru.seqOpt().has_value()) {
  serialized_size(encoding, size, IDL::DistinctType<const  ::optional::seq_int_t, optional_seq_9int_9t_tag>(stru.seqOpt().value()));
  }
}
```

Reply:
Yes, the size should be 7; I forgot about the delimiter in front of the members.

The generated code looks alright to me, though I'll need to remind myself of how `IDL::DistinctType` works.